python/cuml/dask/linear_model/linear_regression.py

```diff
@@ -14,547 +13,315 @@
-                yield part  # trigger error locally
-
-        # A dict in the form of { part_key: part }
-        key_to_part_dict = dict([(str(part.key), part) for part in parts])
-
-        who_has = yield client.who_has(parts)
-
-        worker_parts = {}
-        for key, workers in who_has.items():
-            worker = parse_host_port(first(workers))
-            if worker not in worker_parts:
-                worker_parts[worker] = []
-            worker_parts[worker].append(key_to_part_dict[key])
-
-        """
-        Create IP Handles on each worker hosting input data
-        """
-
-        # Format of input_devarrays = ([(X, y)..], dev)
-        input_devarrays = [(worker, client.submit(fit_to_device_arrays,
-                                                  part, workers=[worker]))
-                           for worker, part in worker_parts.items()]
-
-        yield wait(input_devarrays)
-
-        """
-        Gather IPC handles for each worker and call _fit() on each worker
-        containing data.
-        """
-
-        # Last worker is the only one that can have less items.
-        exec_node = loc_dict[X_df.npartitions-1]
+        Fit the model with X and y.
 
-        # Need to fetch parts on worker
-        on_worker = list(filter(lambda x: x[0] == exec_node, input_devarrays))
-        not_on_worker = list(filter(lambda x: x[0] != exec_node,
-                                    input_devarrays))
-
-        ipc_handles = [client.submit(get_input_ipc_handles, future,
-                                     workers=[a_worker])
-                       for a_worker, future in not_on_worker]
-
-        raw_arrays = [future for a_worker, future in on_worker]
-
-        # IPC Handles are loaded in separate threads on worker so they can be
-        # used to make calls through cython
-        # Calls _fit_on_worker defined in the bottom
-        intercept = client.submit(_fit_on_worker, (ipc_handles, raw_arrays),
-                                  self._build_params_map(),
-                                  workers=[exec_node])
-
-        yield wait(intercept)
-
-        coef_series = [client.submit(coef_on_worker, coefs[i], i,
-                                     X_df.shape[1],
-                                     X_df.npartitions, loc_dict[i],
-                                     workers=[loc_dict[i]])
-                       for i in range(len(loc_dict))]
-
-        # coef_on_worker(self, coef, locations, ncols, nparts, worker):
-
-        raise gen.Return((coef_series, intercept, loc_dict))
-
-    def predict(self, X):
-        """
-        Predict values for the multi-gpu linear regression model by making
-        calls to the predict function with dask-cudf objects.
-
-        :param df:
-            a dask-cudf with data distributed one worker per GPU
-        :return:
-            a dask-cudf containing outputs of the linear regression
+        Parameters
+        ----------
+        X : dask cuDF input
```

Comment:
Docstrings should be formatted as:
```
X : dask cuDF dataframe (n_rows, n_features)
  Features for regression
y : dask cuDF (n_rows, 1)
  Labels (outcome values)
```
 with type on first line and explanation on second

Reply:
Done.